module/VuFind/src/VuFind/Search/Solr/Options.php

```diff
@@ -183,58 +115,38 @@ public function __construct(\VuFind\Config\PluginManager $configLoader)
             = (array)($this->facetSettings['SpecialFacets']['hierarchicalFacetSortOptions'] ?? []);
 
         // Load Spelling preferences
-        $config = $configLoader->get($this->mainIni);
-        if (isset($config->Spelling->enabled)) {
-            $this->spellcheck = $config->Spelling->enabled;
+        if (null !== ($spellcheck = $this->mainConfig['Spelling']['enabled'] ?? null)) {
+            $this->spellcheck = $spellcheck;
         }
 
         // Turn on first/last navigation if configured:
-        if (
-            isset($config->Record->first_last_navigation)
-            && $config->Record->first_last_navigation
-        ) {
+        if ($this->mainConfig['Record']['first_last_navigation'] ?? false) {
             $this->recordPageFirstLastNavigation = true;
         }
 
-        // Turn on highlighting if the user has requested highlighting or snippet
-        // functionality:
-        $highlight = $searchSettings->General->highlighting ?? false;
-        $snippet = $searchSettings->General->snippets ?? false;
+        // Turn on highlighting if the user has requested highlighting or snippet functionality:
+        $highlight = $this->searchSettings['General']['highlighting'] ?? false;
+        $snippet = $this->searchSettings['General']['snippets'] ?? false;
         if ($highlight || $snippet) {
             $this->highlight = true;
         }
 
         // Load autocomplete preferences:
-        $this->configureAutocomplete($searchSettings);
+        $this->configureAutocomplete($this->searchSettings);
 
         // Load shard settings
-        if (
-            isset($searchSettings->IndexShards)
-            && !empty($searchSettings->IndexShards)
-        ) {
-            foreach ($searchSettings->IndexShards as $k => $v) {
-                $this->shards[$k] = $v;
-            }
+        $this->shards = (array)($this->searchSettings['IndexShards'] ?? []);
+        if ($this->shards) {
             // If we have a default from the configuration, use that...
-            if (
-                isset($searchSettings->ShardPreferences->defaultChecked)
-                && !empty($searchSettings->ShardPreferences->defaultChecked)
-            ) {
-                $defaultChecked
-                    = is_object($searchSettings->ShardPreferences->defaultChecked)
-                    ? $searchSettings->ShardPreferences->defaultChecked->toArray()
-                    : [$searchSettings->ShardPreferences->defaultChecked];
-                foreach ($defaultChecked as $current) {
-                    $this->defaultSelectedShards[] = $current;
-                }
+            if ($defaultShards = $this->searchSettings['ShardPreferences']['defaultChecked'] ?? null) {
+                $this->defaultSelectedShards = $defaultShards;
```

Comment:
Should we cast this in case there's a single value instead of an array?

```suggestion
                $this->defaultSelectedShards = (array)$defaultShards;
```

Reply:
Done.